src/Guesser/FilterTypeGuesser.php

```diff
@@ -73,36 +82,46 @@ public function guessType($class, $property, ModelManagerInterface $modelManager
         $options['field_name'] = $metadata->fieldMappings[$propertyName]['fieldName'];
 
         switch ($metadata->getTypeOfField($propertyName)) {
-            case 'boolean':
+            case Type::BOOL:
+            case Type::BOOLEAN:
                 $options['field_type'] = BooleanType::class;
                 $options['field_options'] = [];
 
-                return new TypeGuess('doctrine_mongo_boolean', $options, Guess::HIGH_CONFIDENCE);
-//            case 'datetime':
-//            case 'vardatetime':
-//            case 'datetimetz':
-//                return new TypeGuess('doctrine_orm_datetime', $options, Guess::HIGH_CONFIDENCE);
-//            case 'date':
-//                return new TypeGuess('doctrine_orm_date', $options, Guess::HIGH_CONFIDENCE);
+                return new TypeGuess(BooleanFilter::class, $options, Guess::HIGH_CONFIDENCE);
+            /* @deprecated This type was deprecated since version 3.x, to be removed in 4.0 */
```

Comment:
IMO, these messages should trigger real deprecation errors in order to let the user do the required changes.